SerialPrograms/Source/PokemonSV/Programs/General/PokemonSV_SizeChecker.cpp

```diff
@@ -0,0 +1,264 @@
+ *
+ */
+
+#include "CommonFramework/Exceptions/OperationFailedException.h"
+#include "CommonFramework/InferenceInfra/InferenceRoutines.h"
+#include "CommonFramework/Notifications/ProgramNotifications.h"
+#include "CommonFramework/VideoPipeline/VideoFeed.h"
+#include "CommonFramework/Tools/StatsTracking.h"
+#include "CommonFramework/Tools/VideoResolutionCheck.h"
+#include "NintendoSwitch/Commands/NintendoSwitch_Commands_PushButtons.h"
+#include "Pokemon/Pokemon_Strings.h"
+#include "PokemonSV/Inference/Dialogs/PokemonSV_DialogDetector.h"
+#include "PokemonSV/Inference/Overworld/PokemonSV_OverworldDetector.h"
+#include "PokemonSV/Inference/Boxes/PokemonSV_BoxDetection.h"
+#include "PokemonSV/Inference/Boxes/PokemonSV_BoxEggDetector.h"
+#include "PokemonSV/Programs/Boxes/PokemonSV_BoxRoutines.h"
+#include "PokemonSV_SizeChecker.h"
+
+namespace PokemonAutomation{
+namespace NintendoSwitch{
+namespace PokemonSV{
+
+using namespace Pokemon;
+
+
+SizeChecker_Descriptor::SizeChecker_Descriptor()
+    : SingleSwitchProgramDescriptor(
+        "PokemonSV:SizeChecker",
+        STRING_POKEMON + " SV", "SizeChecker",
+        "ComputerControl/blob/master/Wiki/Programs/PokemonSV/SizeChecker.md",
+        "Check boxes of Pokémon for size marks.",
+        FeedbackType::REQUIRED,
+        AllowCommandsWhenRunning::DISABLE_COMMANDS,
+        PABotBaseLevel::PABOTBASE_12KB
+    )
+{}
+struct SizeChecker_Descriptor::Stats : public StatsTracker{
+    Stats()
+        : m_boxes(m_stats["Boxes Checked"])
+        , m_checked(m_stats["Checked"])
+        , m_empty(m_stats["Empty Slots"])
+        , m_eggs(m_stats["Eggs"])
+    {
+        m_display_order.emplace_back("Boxes Checked");
+        m_display_order.emplace_back("Checked");
+        m_display_order.emplace_back("Empty Slots");
+        m_display_order.emplace_back("Eggs");
+    }
+    std::atomic<uint64_t>& m_boxes;
+    std::atomic<uint64_t>& m_checked;
+    std::atomic<uint64_t>& m_empty;
+    std::atomic<uint64_t>& m_eggs;
+};
+std::unique_ptr<StatsTracker> SizeChecker_Descriptor::make_stats() const{
+    return std::unique_ptr<StatsTracker>(new Stats());
+}
+
+
+
+SizeChecker::SizeChecker()
+    : GO_HOME_WHEN_DONE(false)
+    , BOXES_TO_CHECK(
+        "<b>Number of Boxes to Check:</b>",
+        LockWhileRunning::LOCKED,
+        2, 1, 32
+    )
+    , NOTIFICATIONS({
+        &NOTIFICATION_PROGRAM_FINISH,
+        &NOTIFICATION_ERROR_FATAL,
+    })
+{
+    PA_ADD_OPTION(GO_HOME_WHEN_DONE);
+    PA_ADD_OPTION(BOXES_TO_CHECK);
+    PA_ADD_OPTION(NOTIFICATIONS);
+}
+
+
+
+
+void SizeChecker::enter_check_mode(ConsoleHandle& console, BotBaseContext& context){
+    WallClock start = current_time();
+
+    while (true){
+        if (current_time() - start > std::chrono::minutes(5)){
+            throw OperationFailedException(
+                ErrorReport::SEND_ERROR_REPORT, console,
+                "enter_check_mode(): Failed to enter box mode after 5 minutes.",

```

Comment:
Why does the timeout need to be that huge ?

Reply:
Reduced to 2 min.